ColumnMode/Program.cpp

```diff
@@ -2188,6 +2209,62 @@ void OnPrint(WindowHandles windowHandles)
 	VerifyHR(hr);
 }
 
+void OnPluginRescan(WindowHandles windowHandles, bool skipRescan)
+{
+	 HMENU toplevelMenu = GetMenu(windowHandles.TopLevel);
+	 HMENU pluginMenu = GetSubMenu(toplevelMenu, 3);//Third menu in the list. Should probably just create it in code
+	 int numItems = GetMenuItemCount(pluginMenu);
+	 constexpr int seperatorIndex = 1;
+	 for (int i = numItems; i > seperatorIndex ; i--)
+	 {
+		 HMENU m = (HMENU)GetMenuItemID(pluginMenu, i);
```

Comment:
GetMenuItemID will fail (return -1) in the case where there are no plugins. 
Reason: there are 2 menu items
* Rescan
* Separator
i starts at 2, which isn't a valid item
It might be okay to DestroyMenu an HMENU handle of -1, but it's not good to rely on that. It'd be safer to either change the for loop or to check the return value of GetMenuItemID

Reply:
Nice catch, thanks!